src/npctalk_funcs.cpp

```diff
@@ -207,6 +214,34 @@ void talk_function::do_construction( npc &p )
     p.set_mission( NPC_MISSION_ACTIVITY );
 }
 
+void talk_function::dismount( npc &p )
+{
+    p.npc_dismount();
+}
+
+void talk_function::find_mount( npc &p )
+{
+    // first find one nearby
+    add_msg( "find mount ran " );
+    for( monster &critter : g->all_monsters() ) {
+        add_msg( "all monsters elem " );
+        if( p.can_mount( critter ) ) {
+            add_msg( "found a mount to path to" );
+            p.set_attitude( NPCATT_ACTIVITY );
+            // keep the horse still for some time, so that NPC can catch up to it nad mount it.
+            p.set_mission( NPC_MISSION_ACTIVITY );
+            p.assign_activity( activity_id( "ACT_FIND_MOUNT" ) );
+            p.chosen_mount = &critter;
```

Comment:
What happens when the monster is killed (or otherwise removed from the game) while the NPC moves towards it? The monster pointer stored here is not invalidated than and is still considered non-null, so it will be accessed even through it's invalid now.

I suggest changing `chosen_mount` to be a `weak_ptr`. That way, you can detect when the monster has been removed from the game.